scapy/layers/dns.py

```diff
@@ -1649,23 +1798,52 @@ def make_reply(self, req):
                         continue  # next
                 except TimeoutError:
                     pass
-            # Error
-            break
+            # Still no answer.
+            if self.mDNS:
+                # "Any time a responder receives a query for a name for which it
+                # has verified exclusive ownership, for a type for which that name
+                # has no records, the responder MUST respond asserting the
+                # nonexistence of that record using a DNS NSEC record [RFC4034]."
+                ans.append(DNSRRNSEC(
+                    # RFC6762 sect 6.1 - Negative Response
+                    ttl=self.ttl,
+                    rrname=rq.qname,
+                    nextname=rq.qname,
+                    typebitmaps=RRlist2bitmap([rq.qtype]),
+                ))
```

Comment:
It's a nice touch! According to https://datatracker.ietf.org/doc/html/rfc6762#section-6.1 it should go to the additional record section
> the
   responder MAY also include an NSEC record in the Additional Record
   Section indicating the nonexistence of other rrtypes for that name
   and rrclass

Reply:
I agree but Windows put it in the answers, so I copied that behavior

Reply:
Got it. In practice it should work either way. Though I think I saw libraries that can't handle additional SRV and TXT records described in https://datatracker.ietf.org/doc/html/rfc6763#section-12 when they are put in the answer section.